src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectlyTests.cs

```diff
@@ -0,0 +1,658 @@
+    {
+        throw new NotImplementedException();
+    }
+
+    public static int operator <(Base left, int right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static int operator >(Base left, int right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static int operator <=(Base left, int right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static int operator >=(Base left, int right)
+    {
+        throw new NotImplementedException();
+    }
+}
+
+class Derived : Base, IComparisonOperators<{|#0:Base|}, int, int> // The 'IComparisonOperators<TSelf, TOther, TResult>' interface requires the 'TSelf' type parameter to be filled with the derived type 'Derived'
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("IComparisonOperators<TSelf, TOther, TResult>", "TSelf", "Derived")).RunAsync();
+        }
+
+        [Fact]
+        public async Task BaseUsedDerivedTypeAsTypeParameter()
+        {
+            await PopulateTestCs(@"
+using System;
+using System.Numerics;
+
+class Base : IBitwiseOperators<{|#0:Derived|}, int, int> // The 'IBitwiseOperators<TSelf, TOther, TResult>' interface requires the 'TSelf' type parameter to be filled with the derived type 'Base'
+{
+    static int IBitwiseOperators<Derived, int, int>.operator ~(Derived value)
+    {
+        throw new NotImplementedException();
+    }
+
+    static int IBitwiseOperators<Derived, int, int>.operator &(Derived left, int right)
+    {
+        throw new NotImplementedException();
+    }
+
+    static int IBitwiseOperators<Derived, int, int>.operator |(Derived left, int right)
+    {
+        throw new NotImplementedException();
+    }
+
+    static int IBitwiseOperators<Derived, int, int>.operator ^(Derived left, int right)
+    {
+        throw new NotImplementedException();
+    }
+}
+
+class Derived : Base, IBitwiseOperators<Derived, int, int>
+{ }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForInterface).WithLocation(0).WithArguments("IBitwiseOperators<TSelf, TOther, TResult>", "TSelf", "Base")).RunAsync();
+        }
+
+        [Fact]
+        public async Task ParentClassImplementedIParsableShouldWarn()
+        {
+            await PopulateTestCs(@"
+using System;
+using System.Numerics;
+
+class Foo<TMe> : IDecrementOperators<TMe> where TMe : IDecrementOperators<TMe>
+{
+    static TMe IDecrementOperators<TMe>.operator --(TMe value)
+    {
+        throw new NotImplementedException();
+    }
+}
+class WrongImplementation : Foo<{|#0:int|}> { } // The 'Foo<TMe>' class requires the 'TMe' type parameter to be filled with the derived type 'WrongImplementation' 
+
+class CorrectImplementation : Foo<CorrectImplementation> { }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMIRuleForClass).WithLocation(0).WithArguments("Foo<TMe>", "TMe", "WrongImplementation")).RunAsync();
```

Comment:
Assume we would like to flag and show corresponding message, i.e. when the pattern applied to a class and if the  derived types from that class are not using the correct parameter  show `The '{0}' class requires the '{1}' type parameter to be filled with the derived type '{2}'` instead of `The '{0}' interface requires the '{1}' type parameter to be filled with the derived type '{2}'`